apps/sms/test/marionette/drafts_test.js

```diff
@@ -27,95 +29,135 @@ marionette('Messages Drafts', function() {
+        (draft.subject || draft.shouldHaveAttachment ? '' : draft.text);
+    });
+
+    return conversationWithDraft;
   }
 
-  function assertDraft(draft) {
-    var conversation = messagesApp.Inbox.firstConversation;
-    assert.ok(conversation.getAttribute('class').indexOf('draft') !== -1);
+  function discardDraft(view, draft) {
+    if (draft.text || draft.shouldHaveAttachment) {
+      view.clearMessage();
+    }
+
+    if (draft.subject) {
+      view.hideSubject();
+    }
+
+    if (draft.recipients) {
+      view.clearRecipients();
+    }
+
+    var inboxView = view.backToInbox();
+
+    // Check that draft is discarded and removed from inbox view.
+    var originalConversation = null;
+    client.waitFor(function() {
+      var conversations = inboxView.conversations;
+
+      // In discard draft case we expect exactly one conversation in inbox.
+      if (conversations.length !== 1) {
+        return false;
+      }
+
+      originalConversation = conversations[0];
 
-    // Navigate to drafts (it may be either Conversation or Composer)
-    conversation.tap();
-    messagesApp.Conversation.waitToAppear();
+      return !originalConversation.isDraft && !originalConversation.hasDraft;
+    });
+
+    return originalConversation;
+  }
 
+  function assertDraft(view, draft) {
     // Check that recipient is loaded from draft.
     if (draft.recipients) {
-      var recipients = messagesApp.Composer.recipients;
+      var recipients = view.recipients;
       assert.equal(recipients.length, draft.recipients.length);
 
       draft.recipients.forEach(function(recipient, index) {
-        assert.equal(recipients[index].text(), recipient);
+        assert.equal(recipients[index], recipient);
       });
     }
 
     // Check that text content is loaded.
     if (draft.text) {
-      assert.equal(messagesApp.Composer.messageInput.text().trim(), draft.text);
+      assert.equal(view.messageText, draft.text);
     }
 
     // Check that subject is loaded.
     if (draft.subject) {
-      assert.isTrue(messagesApp.Composer.subjectInput.displayed());
-      assert.equal(messagesApp.Composer.subjectInput.text(), draft.subject);
+      assert.isTrue(view.isSubjectVisible());
+      assert.equal(view.subject, draft.subject);
+    } else {
+      assert.isFalse(view.isSubjectVisible());
     }
 
     // Check that attachment is loaded.
     if (draft.shouldHaveAttachment) {
-      assert.ok(messagesApp.Composer.attachment);
+      assert.equal(view.attachments.length, 1);
     }
 
-    // Check that focus is in composer
-    assertIsFocused(
-      messagesApp.Composer.messageInput, 'Message input should be focused'
+    assert.isTrue(
+      view.isMessageInputFocused(), 'Message input should be focused'
     );
   }
 
+  function assertConversation(conversation, properties) {
+    Object.keys(properties).forEach(function(propertyKey) {
+      assert.equal(conversation[propertyKey], properties[propertyKey]);
+    });
```

Comment:
We have ```assert.deepEqual```, I think initially I wanted to just compare subset of fields while ```assert.deepEqual``` requires all fields to be presented. But maybe it's a good to assert every field anyway - will change, thanks!